src/pages/contact.js

```diff
@@ -0,0 +1,185 @@
+/** @jsx jsx */
+import { useCallback, useState, useEffect } from 'react'
+import { Redirect } from '@reach/router'
+import {
+  Button,
+  Text,
+  Box,
+  Label,
+  Input,
+  Select,
+  Textarea,
+  jsx,
+} from 'theme-ui'
+
+import Layout from '../components/layout'
+import PageHeader from '../components/pageheader'
+import Container from '../components/container'
+
+import firebase from 'firebase'
+import 'firebase/firestore'
+import getFirebase from '../firebase'
+
+export default function login() {
+  const firebaseApp = getFirebase()
+
+  const [error, setError] = useState(null)
+  const [success, setSuccess] = useState(null)
+
+  const [name, setName] = useState('')
+  const handleSetName = useCallback(e => setName(e.target.value), [name])
+
+  const [email, setEmail] = useState('')
+  const handleSetEmail = useCallback(e => setEmail(e.target.value), [email])
+
+  const [type, setType] = useState('Student')
+  const handleSetType = useCallback(e => setType(e.target.value), [type])
+
+  const [comment, setComment] = useState('')
+  const handleSetComment = useCallback(e => setComment(e.target.value), [
+    comment,
+  ])
+
+  const [formSubmit, setFormSubmit] = useState(false)
+
+  const submitContactForm = useCallback(
+    e => {
+      e.preventDefault()
+
+      setError('')
+
+      if (name === '' || email === '' || type === '' || comment === '') {
+        setError('One or more fields are missing.')
+        return
+      }
+
+      if (!/\S+@\S+\.\S+/.test(email)) {
+        setError('Invalid email.')
+        return
+      }
+
+      firebaseApp
+        .firestore()
+        .collection('contact')
+        .add({
+          name: name,
+          email: email,
+          type: type,
+          comment: comment,
+        })
+        .then(() => {
+          setSuccess('Sucessfully submitted contact form!')
+          setTimeout(setFormSubmit.bind(null, true), 1500)
```

Comment:
?

Reply:
shows success message, waits 1.5s, the redirects to /

Reply:
This would likely be handled better as an imperative navigation instead of rendering a `Redirect` component.